### PR TITLE
v12 embryonal subtyping (7/N)

### PR DESCRIPTION
**Pull request by ewafula**
#### What scientific question is your analysis addressing?
This PR runs Embryonal subtyping for v12

#### What was your approach?
made the following changes:
- Updated script 01 to include high-confidence methylation embryonal subtypes from the histologies base file where subtypes are either `NA` or `CNS Embryonal, NOS`
- Checked to confirm that all gene symbols utilized within the module  are up to date with Gencode v39
```
NEW	OLD
TTYH1   TTYH1
MN1     MN1
FOXR2   FOXR2
CIC     CIC
PATZ1   PATZ1
LIN28A  LIN28A
BCOR    BCOR
HGNET   HGNET
```
- README updated.

#### What GitHub issue does your pull request address?
Part of the v12 release

#### Which areas should receive a particularly close look?
Please review scripts 01 and 04 in detail to ensure the logic is correct.

#### Is there anything that you want to discuss further?
NA

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
embryonal subtyping table output.

#### What is your summary of the results?
- embryonal_tumor_molecular_subtypes.tsv

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by kelseykeith** [comment]
Since this review was an ASAP request, just want to comment with a quick update. Code runs reproducibly and error-free and for samples that were subtyped both in v11 and this v12 rerun, subtypes match. However, there's only 8 samples that overlap between v11 and v12 so I'm just running down the differences there and then I will either request changes or approve.

**Comment by kelseykeith** [comment]
Never mind about the 8 samples, I was comparing the wrong files. Everything looks good:
- code runs correctly
- includes the same samples as v11 plus additional samples
- correctly replaces 'CNS Embryonal, NOS' with the methylation molecular subtype if available

**Review by kelseykeith — APPROVED** [review]

**Review by rjcorb — APPROVED** [review]
LGTM! everything runs fine and logic is correct.

**Comment by ewafula** [comment]
@jharenza, the following BCOR and LGG tumor samples are no longer being subtyped in as embryonal following your suggested code logic updates. 

CNS_BCOR‐ITD) tumors
```
# A tibble: 3 x 2
  Kids_First_Biospecimen_ID short_histologies   
  <chr>                     <chr>               
1 BS_21KG9DWJ               Embryonal tumor, EWS
2 BS_P9G0E57D               Embryonal tumor, EWS
3 BS_XQNMWJ4H               Embryonal tumor, EWS
```

Low-grade glioma/astrocytoma tumors
```
# A tibble: 4 x 2
  Kids_First_Biospecimen_ID short_histologies    
  <chr>                     <chr>                
1 BS_CVGFNCBJ               Embryonal tumor, LGAT
2 BS_SD28E0SC               Embryonal tumor, LGAT
3 BS_WSFB4SA2               Embryonal tumor, LGAT
4 BS_YCREDZTJ               Embryonal tumor, LGAT
```

**Comment by ewafula** [comment]
`BS_Z62B53CW` no longer subtyped ETMR.
```
# A tibble: 1 x 2
  Kids_First_Biospecimen_ID short_histologies
  <chr>                     <chr>            
1 BS_Z62B53CW               ETMR, Meningioma
```

File: analyses/molecular-subtyping-embryonal/01-samples-to-subset.Rmd [hunk 1]
```diff
@@ -224,6 +255,54 @@ histologies_df %>%
...
+                !Kids_First_Biospecimen_ID %in% unique(emb_methyl_subtyped$Kids_First_Biospecimen_ID_Methyl),
+                experimental_strategy == "Methylation", 
+                sample_type == "Tumor", 
+                composition == "Solid Tissue",
+                cohort != "TARGET") %>%
+  dplyr::left_join(emb_methyl_subtyped %>% dplyr::select(Kids_First_Participant_ID, sample_id, molecular_subtype_methyl),
+                   by = c("Kids_First_Participant_ID", "sample_id")) %>%  
+  dplyr::mutate(molecular_subtype_methyl = case_when(
+    is.na(molecular_subtype_methyl) ~ "CNS Embryonal, NOS",
+    TRUE ~ molecular_subtype_methyl)) %>%
```
Comment (jharenza) [inline analyses/molecular-subtyping-embryonal/01-samples-to-subset.Rmd]:
```suggestion
  dplyr::mutate(molecular_subtype_methyl = case_when(
    is.na(molecular_subtype_methyl) ~ NA_character,
    TRUE ~ molecular_subtype_methyl)) %>% 
```

Comment (jharenza) [reply analyses/molecular-subtyping-embryonal/01-samples-to-subset.Rmd]: We don't want to create new methylation subtypes - only report high confidence or none

**Pull request merged**